src/main/scala/za/co/absa/hyperdrive/trigger/api/rest/services/KafkaService.scala

```diff
@@ -0,0 +1,80 @@
+/*
+ * Copyright 2018 ABSA Group Limited
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package za.co.absa.hyperdrive.trigger.api.rest.services
+
+import org.apache.kafka.clients.consumer.{ConsumerConfig, KafkaConsumer}
+import org.apache.kafka.common.TopicPartition
+import org.springframework.stereotype.Service
+import za.co.absa.hyperdrive.trigger.api.rest.services.KafkaServiceImpl.{BeginningOffsets, EndOffsets, OffsetFunction}
+import za.co.absa.hyperdrive.trigger.configuration.application.{GeneralConfig, KafkaConfig}
+
+import java.util.Properties
+import java.util.concurrent.ConcurrentHashMap
+import javax.inject.Inject
+import scala.collection.JavaConverters._
+
+trait KafkaService {
+  def getBeginningOffsets(topic: String, consumerProperties: Properties): Map[Int, Long]
+  def getEndOffsets(topic: String, consumerProperties: Properties): Map[Int, Long]
+}
+
+@Service
+class KafkaServiceImpl @Inject() (kafkaConfig: KafkaConfig, generalConfig: GeneralConfig) extends KafkaService {
+
+  private val kafkaConsumersCache = new ConcurrentHashMap[Properties, KafkaConsumer[String, String]]()
+
+  override def getBeginningOffsets(topic: String, consumerProperties: Properties): Map[Int, Long] = {
+    getOffsets(topic, consumerProperties, BeginningOffsets)
+  }
+
+  override def getEndOffsets(topic: String, consumerProperties: Properties): Map[Int, Long] = {
+    getOffsets(topic, consumerProperties, EndOffsets)
+  }
+
+  def createKafkaConsumer(properties: Properties) = new KafkaConsumer[String, String](properties)
+
+  private def getOffsets(topic: String, properties: Properties, offsetFn: OffsetFunction): Map[Int, Long] = {
+    val groupId = s"${kafkaConfig.groupIdPrefix}-${generalConfig.appUniqueId}-kafkaService"
```

Comment:
I think it is not necessary to have kafka and general config here, just any uuid should be fine.